src/main/java/com/linecorp/armeria/client/manipulating/HttpHeaderManipulatingClientBuilder.java

```diff
@@ -0,0 +1,197 @@
+     *                  When generator returns null or empty value, the returned value will be ignored.
+     *
+     * @return this instance
+     *
+     * @throws NullPointerException when header or generator function is null
+     * @throws IllegalArgumentException when header is empty
+     */
+    public HttpHeaderManipulatingClientBuilder set(AsciiString header,
+                                                   Function<String, String> generator) {
+        requireNotEmpty(header, "header");
+        requireNonNull(generator, "generator should not be null");
+        return appendOperator(request -> {
+            String oldValue = request.headers().get(header);
+            String newValue = generator.apply(oldValue);
+            if (newValue != null && !newValue.isEmpty()) {
+                request.headers().set(header, newValue);
+            }
+        });
+    }
+
+    /**
+     * Appends an remove operation that removes given header from request.
+     * @param header header to remove
+     *
+     * @return this instance
+     *
+     * @throws NullPointerException when header is null
+     * @throws IllegalArgumentException when header is empty
+     */
+
+    public HttpHeaderManipulatingClientBuilder remove(AsciiString header) {
+        requireNotEmpty(header, "header");
+        return appendOperator(request -> request.headers().remove(header));
+    }
+
+    /**
+     * Appends an conditional remove operation that removes all (header, value) entries passes the test
+     * of given predicate.
+     *
+     * @param predicate function that returns true when input (header, value) is to be removed.
+     * @return this instance
+     *
+     * @throws NullPointerException when predicate is null
+     */
+    public HttpHeaderManipulatingClientBuilder remove(BiPredicate<AsciiString, String> predicate) {
+        requireNonNull(predicate, "predicate should not be null");
+        return appendOperator(request -> {
+            HttpHeaders headers = request.headers();
+            headers.forEach(entry -> {
+                AsciiString header = entry.getKey();
+                if (predicate.test(header, entry.getValue())) {
+                    headers.remove(header);
+                }
+            });
+        });
+    }
+
+    /**
+     * Appends an update operation to manipulator to do some complex operation that cannot be done with
+     * other simple operations.
+     *
+     * @param updater function that updates given request headers.
+     * @return this instance
+     * @throws NullPointerException when updater is null
+     */
+    public HttpHeaderManipulatingClientBuilder update(Consumer<HttpHeaders> updater) {
+        requireNonNull(updater, "updater should not be null");
+        return appendOperator(request -> updater.accept(request.headers()));
+    }
+
+    private static void requireNotEmpty(CharSequence str, String target) {
+        requireNonNull(str, target + " should not be null");
```

Comment:
`target` is enough for the message. Please check the convention for Armeria. https://github.com/line/armeria/search?utf8=%E2%9C%93&q=requireNonNull